data-prepper-plugins/rss-source/src/test/java/org/opensearch/dataprepper/plugins/source/rss/RSSSourceTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rss;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class RSSSourceTest {
+
+    private final String PLUGIN_NAME = "rss";
+
+    private final String PIPELINE_NAME = "test";
+
+    private final String VALID_RSS_URL = "https://forum.opensearch.org/latest.rss";
+
+    private final String INVALID_RSS_URL = "https://docs.aws.amazon.com/amazondynamodb/latest";
+
+    private PluginMetrics pluginMetrics;
+
+    private RSSSourceConfig rssSourceConfig;
+
+    private RSSSource rssSource;
+
+    @BeforeEach
+    void setUp() {
+        pluginMetrics = PluginMetrics.fromNames(PLUGIN_NAME, PIPELINE_NAME);
+        rssSourceConfig = mock(RSSSourceConfig.class);
+        rssSource = mock(RSSSource.class);
+    }
+
+    @Test
+    void test_when_extractItemsFromRssFeed_captured() {
+        ArgumentCaptor<RSSSourceConfig> argumentCaptor = ArgumentCaptor.forClass(RSSSourceConfig.class);
+        when(rssSourceConfig.getUrl()).thenReturn(VALID_RSS_URL);
+        doNothing().when(rssSource).extractItemsFromRssFeed(argumentCaptor.capture());
```

Comment:
Not blocking for this PR. But `extractItemsFromRssFeed` should become private method once the writing into buffer logic is in-place. So I suppose these tests will need be modified later